sphinx/texinputs/sphinxlatexshadowbox.sty

```diff
@@ -113,4 +148,31 @@
    \spewnotes
   }
 
+% 8.1.0
+\newenvironment{sphinxtopic}
+               {\begin{sphinxShadowBox}\relax}{\end{sphinxShadowBox}}
+\newenvironment{sphinxcontents}
+               {\begin{sphinxShadowBox}[contents]}{\end{sphinxShadowBox}}
+% Arguably sphinxsidebar should rather use a wrapfig or similar environment
+% but this is so dysfunctional in LaTeX (except for self-written documents)
+% so we prefer to not venture into such a potential quagmire and keep the
+% legacy rendering using a full width display.
+\newenvironment{sphinxsidebar}
+               {\begin{sphinxShadowBox}[sidebar]}{\end{sphinxShadowBox}}
+
+% TODO: decide if this should be in sphinxlatexstyletext.sty rather
+%
+% 8.1.0 styles topic/contents/sidebar with a title row, too.
+% Prior to 8.1.0, definitions use \protected\def but there does not seem
```

Comment:
I think you are both right and wrong.
- using `\protected\def` is surely faster at definition time compared to `\newcommand`,
- at execution time it will be less efficient bcause TeX will check for argument not to contain an explicit `\par`; this is reason why LaTeX3 basically defines everything `\long` (and also probably because they can not hook into core TeX error and the error message is not enlightening to users, especially as `\par` arises from empty lines which people do not know).

+1 for `\newcommand*` (contradicting my opinion in previous paragraph about interest to user) but only if I really convince myself the title has to be from Docutils only one paragraph. I am not so sure right now, although I may have test it, and besides `\protected\def` would not have accepted it so perhaps I did check in the past.  But you are definitely right the more direct analog is `\newcommand*`, as the  `\protected` is not likely at all to have any usefulness, such stuff not being in "moving arguments".

Reply:
This business of `\newcommand` is also related to my visual comfort in my Emacs buffers using AUCTeX.  There will be some syntax highlighting which is much better than with `\def`.... it helps me focus on the user level interface...

Reply:
After checking [Docutils](https://docutils.sourceforge.io/docs/ref/rst/directives.html#topic) it seems a topic title (or contents title, or sidebar title) can never occupy more than one paragraph -- this is not so clear from the Docutils doc, but is observed in practice.  So we could use `\newcommand*`, but why if it only adds overhead to execution time...

Reply:
> at execution time it will be less efficient bcause TeX will check for argument not to contain an explicit \par

Ah right. Well, we can keep `\newcommand` even though we know that we won't have `\par` inside (and to speed-up a bit the execution time), or we keep `\newcommand*` to be consistent with `\protected\def` but we slow down a bit (well it'll be as slow as the `\protected\def` I think, maybe a bit more since there are some expansions behind the scenes IIRC).

Reply:
I will push a commit with `\newcommand*`.  Ultimately the command argument is, per default, given over to `\sphinxdotitlerow` which itself passes it to  `\sphinxstrong` which is defined via `\protected\def`.

By the way all LaTeX commands such as `\textbf` are "short" i.e. as if defined with `\newcommand*`.

The `\protected` in `\protected\def` is used as a general rule in `sphinxlatexstyletext.sty` because the styling macros may generally occur in titles or captions, which may cause problems due to their expansion when being written to auxiliary files.

The `\sphinxstyletopictitle` et al. were initially put into `sphinxlatexstyletext.sty` for lack of a better place (now moved to ``sphinxlatexshadowbox.sty``) and were "contaminated" by `\protected\def`.  But they can not occur in such "moving arguments", so there is no strong reason to use `\protected` (or `\DeclareRobusCommand`).

However there is indeed rationale for `\def`, i.e. making them "short" macros, with no regrets as it seems anyhow per Docutils syntax rules it can never be multi-paragraph (I find Docutils doc not clear, but this is what I observe).

Pushing `\newcommand*` next.